draft-ietf-quic-transport.md

```diff
@@ -4067,9 +4065,13 @@ Final Offset:
 
 ## CONNECTION_CLOSE frame {#frame-connection-close}
 
-An endpoint sends a CONNECTION_CLOSE frame (type=0x02) to notify its peer that
-the connection is being closed.  CONNECTION_CLOSE is used to signal errors at
-the QUIC layer, or the absence of errors (with the NO_ERROR code).
+An endpoint sends a CONNECTION_CLOSE frame (type=0x02 or 0x03) to notify its
+peer that the connection is being closed.  The CONNECTION_CLOSE with a frame
+type of 0x02 is used to signal errors at the QUIC layer, or the absence of
+errors (with the NO_ERROR code).  The CONNECTION_CLOSE frame with a type of 0x03
+is used to signal an error with the protocol that uses QUIC.  Protocols that use
+QUIC only use the CONNECTION_CLOSE frame of type 0x03; the frame with type 0x02
+is only used internally by QUIC.
```

Comment:
More inclined to indicate this as an embedded flag, indicating whether the error is at the transport or application layer.  Also makes the optional field easier.

Reply:
As I said elsewhere, that's a wash when you try it out.